macros/src/codegen.rs

```diff
@@ -1913,52 +1916,45 @@ fn mk_typenum_capacity(capacity: u8, power_of_two: bool) -> proc_macro2::TokenSt
     quote!(rtfm::export::consts::#ident)
 }
 
-fn mk_ident(name: Option<&str>) -> Ident {
-    static CALL_COUNT: AtomicUsize = AtomicUsize::new(0);
-
-    let elapsed = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
-
-    let secs = elapsed.as_secs();
-    let nanos = elapsed.subsec_nanos();
-
-    let count = CALL_COUNT.fetch_add(1, Ordering::SeqCst) as u32;
-    let mut seed: [u8; 16] = [0; 16];
-
-    for (i, v) in seed.iter_mut().take(8).enumerate() {
-        *v = ((secs >> (i * 8)) & 0xFF) as u8
-    }
-
-    for (i, v) in seed.iter_mut().skip(8).take(4).enumerate() {
-        *v = ((nanos >> (i * 8)) & 0xFF) as u8
-    }
+struct IdentGenerator {
+    rng: rand::rngs::SmallRng,
+}
 
-    for (i, v) in seed.iter_mut().skip(12).enumerate() {
-        *v = ((count >> (i * 8)) & 0xFF) as u8
+impl IdentGenerator {
+    fn new() -> IdentGenerator {
+        let crate_name = env!("CARGO_PKG_NAME");
+        let seed = [0u8; 16];
+        for (i, b) in crate_name.bytes().enumerate() {
+            seed[i%seed.len()].wrapping_add(b);
+        }
+        IdentGenerator { rng: rand::rngs::SmallRng::from_seed(seed) }
     }
 
-    let n;
-    let mut s = if let Some(name) = name {
-        n = 4;
-        format!("{}_", name)
-    } else {
-        n = 16;
-        String::new()
-    };
-
-    let mut rng = rand::rngs::SmallRng::from_seed(seed);
-    for i in 0..n {
-        if i == 0 || rng.gen() {
-            s.push(('a' as u8 + rng.gen::<u8>() % 25) as char)
+    fn mk_ident(&mut self, name: Option<&str>) -> Ident {
+        let n;
+        let mut s = if let Some(name) = name {
+            n = 4;
+            format!("{}_", name)
         } else {
-            s.push(('0' as u8 + rng.gen::<u8>() % 10) as char)
+            let crate_name = env!("CARGO_PKG_NAME").replace("-", "_").to_lowercase();
+            n = 16;
+            format!("{}__internal__", crate_name)
+        };
```

Comment:
If prepending the package name is good enough namespacing, we could just as well append sequentially increasing digits instead of randomly generating a suffix.